ocp_resources/machine_config_pool.py

```diff
@@ -2,8 +2,70 @@
 
 
 class MachineConfigPool(Resource):
+    """
+    MachineConfigPool object. API reference:
+    https://docs.openshift.com/container-platform/4.12/rest_api/machine_apis/machineconfigpool-machineconfiguration-openshift-io-v1.html
+
+    Args:
+        name (string): Object name.
+        machine_config_selector (object): List of matching labels, to determine on which nodes to apply this
+            MachineConfigPool object.
+        configuration (object): Current MachineConfig object for the machine config pool.
+        yaml_file (str): yaml file for the resource.
+        node_selector_labels (object (string)): Map of {key,value} pairs that determine node selection criteria.
+        max_unavailable (integer/string): Number/percentage of nodes that can go Unavailable during an update.
+        paused (object (boolean)): Whether changes to this MachineConfigPool should be stopped.
+    """
+
     api_group = Resource.ApiGroup.MACHINECONFIGURATION_OPENSHIFT_IO
 
     class Status(Resource.Status):
         UPDATED = "Updated"
         UPDATING = "Updating"
+
+    def __init__(
+        self,
+        name=None,
+        machine_config_selector=None,
+        configuration=None,
+        yaml_file=None,
+        node_selector_labels=None,
+        max_unavailable=None,
+        paused=None,
+        **kwargs,
+    ):
+        super().__init__(name=name, yaml_file=yaml_file, **kwargs)
+        self.configuration = configuration
+        self.machine_config_selector = machine_config_selector
+        self.node_selector_labels = node_selector_labels
+        self.max_unavailable = max_unavailable
+        self.paused = paused
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.update(
+                {
+                    "spec": {
+                        "configuration": self.configuration or {},
+                    },
+                },
+            )
+
+            if self.machine_config_selector:
+                self.res["spec"]["machineConfigSelector"] = {}
+                self.res["spec"]["machineConfigSelector"][
+                    "matchExpressions"
+                ] = self.machine_config_selector
+
+            if self.node_selector_labels:
+                self.res["spec"]["nodeSelector"] = {}
+                self.res["spec"]["nodeSelector"][
+                    "matchLabels"
```

Comment:
can also be other things, for example `matchExpressions`  https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements

Reply:
Moved to kwargs, as it is already handled in `Resource`.

Reply:
Ack